src/query/plan/operator.cpp

```diff
@@ -448,7 +448,7 @@ class ScanAllCursor : public Cursor {
 #ifdef MG_ENTERPRISE
   bool FindNextVertex(const ExecutionContext &context) {
     while (vertices_it_.value() != vertices_.value().end()) {
-      if (context.auth_checker->Has(*vertices_it_.value(), memgraph::storage::View::OLD,
+      if (context.auth_checker->Has(*vertices_it_.value(), memgraph::storage::View::NEW,
```

Comment:
I think the proper solution here is to push down the value of `ScanAll::view_` into the `ScanAll` cursor, otherwise we can run into problems with the view. E.g.: the current solution might issues with `MATCH (n), (m) DELETE n`, because the second `ScanAll` might access a node that is deleted by in the same query, thus getting its labels with `View::NEW` will result in a `DELETED_OBJECT` error. I am not sure that this query will trigger, but I am almost sure that something can trigger this error if we are not careful with it.